### ISSUE TITLE
RFE: replace PresentationFramework with something suitable for other platforms.

### ISSUE THREAD

**Issue by qtlin** [i3arnon/YouCast#70]
References
1. https://www.mono-project.com/docs/gui/wpf/ - No plans to implement WPF.
2. https://github.com/yar229/WebDavMailRuCloud - This project has similar structure and works on all platforms.

**Comment by i3arnon** [comment]
I've been thinking about this for quite some time..

YouCast is mostly built using WCF's syndication (for RSS) and WPF. Those are the old windows-only technologies of .NET which isn't surprising considering I made it about a decade ago.
That means making it cross platform requires basically rewriting everything..

I have been looking at it from time to time and the current plan is to use https://github.com/dotnet/SyndicationFeedReaderWriter for RSS in an ASP.NET server and the yet unreleased .NET MAUI for x-plat UI.
Both are needed to make it possible for YouCast to run on non-windows machines.

This is a pretty big investment, so it's hard for me to justify going ahead with it, especially since there's no iterative value until everything is done.

**Comment by i3arnon** [comment]
Would there be value in creating a command-line port of YouCast for other platforms? Without any GUI?

**Comment by qtlin** [comment]
Absolutely. A something like https://github.com/yar229/WebDavMailRuCloud in dotnet is the most convenient, in my personal opinion, cross platform solution. Windows version can use the same --service option to install itself as a service. Linux has its own trivial method to install as a service.

**Comment by qtlin** [comment]
Hello @i3arnon 
Any progress towards pure .NET Core implementation?
